packages/mask/src/plugins/Predict/Azuro/PlaceBetDialog.tsx

```diff
@@ -0,0 +1,235 @@
+import { ConfirmModal } from '../../Tips/components/common/ConfirmModal'
+import { activatedSocialNetworkUI } from '../../../social-network'
+import { SuccessIcon } from '@masknet/icons'
+import { isTwitter } from '../../../social-network-adaptor/twitter.com/base'
+import { isFacebook } from '../../../social-network-adaptor/facebook.com/base'
+import { contractAddresses } from '../constants'
+
+const useStyles = makeStyles()((theme) => ({
+    root: {
+        fontFamily: 'Muli,Helvetica,-apple-system,system-ui,"sans-serif"',
+        width: '100%',
+        boxShadow: 'none',
+        border: `solid 1px ${theme.palette.divider}`,
+        padding: 0,
+    },
+    content: {
+        width: '100%',
+        height: 'var(--contentHeight)',
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'space-around',
+        padding: '0 !important',
+    },
+    walletStatusBox: { margin: theme.spacing(1, 1, 3, 1) },
+    container: { padding: theme.spacing(1) },
+    infosContainer: {
+        padding: theme.spacing(1.75, 0.75, 0.75, 0.75),
+        gap: theme.spacing(0.5),
+    },
+    infoContainer: {
+        minHeight: 30,
+    },
+    infoTitle: {
+        fontWeight: '300',
+    },
+    actionButton: {
+        flexDirection: 'column',
+        position: 'relative',
+        marginTop: theme.spacing(1.5),
+        lineHeight: '22px',
+        justifyContent: 'center',
+        alignItems: 'center',
+        padding: '13px 0',
+        fontSize: 18,
+    },
+}))
+
+export interface PlaceBetDialogProps {
+    open: boolean
+    onClose?: () => void
+    game: AzuroGame
+    condition: Odds
+}
+
+export function PlaceBetDialog(props: PlaceBetDialogProps) {
+    const { open, onClose, game, condition } = props
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const [amount, setAmount] = useState('')
+    const [slippage, setSlippage] = useState<number>(2)
+    const [confirmModalIsOpen, openConfirmModal] = useBoolean(false)
+    const chainId = useChainId()
+    const { AZURO_LP } = useAzuroConstants()
+    const {
+        value: token,
+        error,
+        loading,
+    } = useFungibleToken(NetworkPluginID.PLUGIN_EVM, contractAddresses[chainId as ChainId]?.token)
+    console.log('amount: --------------------- ', amount)
+    console.log('token: --------------------- ', token)
+
+    const { value: balance } = useFungibleTokenBalance(NetworkPluginID.PLUGIN_EVM, token?.address)
+    const { value: actualRate, loading: actualRateLoading } = useActualRate(condition, amount)
+
+    const rawAmount = rightShift(String(amount), USDT_DECIMALS)
+    const deadline = Math.floor(Date.now() / 1000) + 2000
+    const { minRate, loading: minRateLoading } = useMinRate(slippage, actualRate ?? 0)
+    const rawMinRate = rightShift(minRate, RATE_DECIMALS)
+
+    const [transactionState, transactionCallback, resetTransactionCallback] = usePlaceBetCallback(
```

Comment:
`TransactionState` is not used anymore. All transactions should send and confirm in a single function call.

Learn more: https://docs.mask.io/docs/web3/connection

Reply:
Thanks, I was looking for this. I'm currently fixing this.

Reply:
Done.